src/main/java/fi/aalto/cs/apluscourses/intellij/model/APlusProject.java

```diff
@@ -40,43 +48,38 @@ public LibraryTable getLibraryTable() {
     return LibraryTablesRegistrar.getInstance().getLibraryTable(project);
   }
 
-  @NotNull
-  public Path getLibraryPath(String name) {
-    return  Paths.get("lib", name);
-  }
-
-  @NotNull
-  public Path getModulePath(String name) {
-    return Paths.get(name);
-  }
-
-  public int resolveLibraryState(String name) {
-    return resolveComponentState(getLibraryTable().getLibraryByName(name), getLibraryPath(name));
-  }
-
-  public int resolveModuleState(String name) {
-    return resolveComponentState(getModuleManager().findModuleByName(name), getModulePath(name));
-  }
-
-  private int resolveComponentState(@Nullable Object componentObj, @NotNull Path dirPath) {
+  /**
+   * Returns a state in which the given component should be.
+   * @param component An IntelliJ specific component.
+   * @param <T> Type of the component.
+   * @return Component state.
+   */
+  public <T extends Component & IntelliJComponent<?>> int resolveComponentState(
```

Comment:
the methods has a dendroid logics and feels important, can it be somehow unit-tested?

Reply:
With some mocking, yes. I'll do! Perhaps even refactor this to another class, as it feels a bit out of place here, or whaddya think?